packages/cli/src/InternalHooks.ts

```diff
@@ -301,6 +301,11 @@ export class InternalHooks {
 			return;
 		}
 
+		if (runData?.status === 'waiting') {
```

Comment:
perhaps the issue here is that `onWorkflowPostExecute`is being called in the first place. we should prevent that, instead of adding a `return` here.
@krynble opinions?

Reply:
I had the same thought and we discussed this here: https://linear.app/n8n/issue/PAY-1298/bug-log-streaming-fires-failed-workflow-notification-on-long-wait#comment-b8670cbe

I think we can change that when we refactor the workflow life-cycle hooks in general. Let me know what you think.

Reply:
I think for the sake of clarity it's best to call the function and return early rather than not call it, otherwise anyone looking at this function alone would expect it to run always.